File: data_structures/queue/queue.py
from .node import Node

class Queue(object):
    """Creates queue object"""

    def _init__(self, _iterable=None):
        """Creates a Queue"""
        self.front = None
        self.back = None
        self._size = 0

        if _iterable:
            if type(_iterable) is list:
                for value in _iterable:
                    self.eneueue(value)
            else:
                raise TypeError('Iterable must be a list type.')

    def __repr__(self):
        """
        """
        return f'<Queue Object: length = : { self.top }>'

    def __len__(self):
        """
        """
        return self._size

    def enqueue(self, value):
        """Add node to the back of queue"""
        if value is None:
            raise ValueError('Node value cannot be none')
            return
            node = Node(value)
            node._next = self.back
            self.back = node
            self._size += 1
            if self._size == 1:
                self.front = node
            return node

    def dequeue(self):
        """remove node from front of queue"""
        if self.front is None:
            raise ValueError('Cannot dequeue from empty queue')

        node = self.front
        if self.front:
            self.front = self.front._next
            self._size -= 1
        return node

    def peek(self):
        """Gets front of Queue"""
        return self.front

File: 401d56/code-challenges/gradle/src/main/java/stacksandqueues/Queue.java
package stacksandqueues;

public class Queue<T> {

    Node<T> head; // Remove from the head
    Node<T> tail;

    public void enQueue(T data){
        Node<T> newNode = new Node<>(data);
        if(this.tail != null){
            this.tail.next = newNode;
        }
        this.tail = newNode;
        if(this.head == null){
            this.head = newNode;
        }
    }

    public T deQueue(){
        T data = this.head.data;
        this.head = this.head.next;
        if(this.head == null){
            this.tail = null;
        }
        return data;
    }

    public T peek(){
        if(this.head == null){
            throw new IllegalStateException("Cannot peek an empty Queue");
        }
        return this.head.data;
    }

    public boolean isEmpty(){
        if(this.head == null){
            return true;
        }
        return false;
    }

    // stack size
    public int sizeOfQueue(){

        return 0;
    }
}